include/fmt/ranges.h

```diff
@@ -390,22 +390,36 @@ using range_formatter_type = conditional_t<
 template <typename R>
 using maybe_const_range =
     conditional_t<has_const_begin_end<R>::value, const R, R>;
+
+template <typename R>
+struct is_not_recursive_range : bool_constant<
+    !std::is_same<detail::uncvref_type<R>, R>::value> {};
```

Comment:
Contrary to the name of this trait it only checks recursiveness but not whether R is a range. I suggest adding is_range check here.

Reply:
Again, we want to prevent instantiation if the is_recursive predicate *unless* the is_range predicate has evaluated to true. See above. We could rename it to something different, such as `is_not_recursive`, if you prefer and add a comment that says that we assume here that is_range is evaluated to true.

Reply:
Yes, please add a comment.

Reply:
Comment added, also relating to [P2286](https://wg21.link/p2286).